lib/LocalModel.js

```diff
@@ -399,88 +399,75 @@ LocalModel.prototype.predictProbability = function (inputData, missingStrategy,
    * @param {object} inputData Input data to predict
    * @param {0|1} missingStrategy Code for the missing strategy to be used
-   * @param {boolean} compact If False, prediction is returned as a
-   *                          list of maps, one per class, with the keys
-   *                          "prediction" and "probability" mapped to the
-   *                          name of the class and it's probability,
-   *                          respectively.  If True, returns a list of
-   *                          probabilities ordered by the sorted order
-   *                          of the class names.
+   * @param {function} cb Callback
    */
   var prediction, self = this;
   if (arguments.length < 2 && (typeof missingStrategy === 'function')) {
     return self.predictProbability(inputData, undefined, missingStrategy);
   }
 
-    function syncPredictProbability(error, prediction) {
-      /**
-       * Creates a local prediction using the model's tree info.
-       *
-       * @param {object} error Error message
-       * @param {object} prediction Regular prediction information.
-       */
-      var output = [], instances = 1.0, k,
+  function syncPredictProbability(error, prediction) {
+    /**
+     * Creates a local prediction using the model's tree info.
+     *
+     * @param {object} error Error message
+     * @param {object} prediction Regular prediction information.
+     */
+    var output = [],
         rootDistribution = self.tree.distribution,
-        categoryMap = {}, distribution = prediction.distribution, index,
-        len = rootDistribution.length, total = 0.0;
+        distribution = self.tree.weighted ? prediction.weightedDistribution : prediction.distribution,
+        categoryMap = {},
+        len = rootDistribution.length,
+        instances = 0.0,
+        total = 0.0,
+        k, index;
 
-      if (error) {
-        return cb(error, null);
-      }
-      if (self.regression) {
-        output = {"prediction": prediction};
-      } else {
-        if (self.tree.weighted) {
-          for (index = 0; index < len; index++) {
-            categoryMap[rootDistribution[index][0]] = 0.0;
-          }
-        } else {
-          // distribution in the root node
-          for (index = 0; index < len; index++) {
-            total += rootDistribution[index][1];
-            categoryMap[rootDistribution[index][0]] = rootDistribution[index][1];
-          }
-          for (index = 0; index < len; index++) {
-            categoryMap[rootDistribution[index][0]] /= total;
-          }
+    if (error) {
+      return cb(error, null);
+    }
+    if (self.regression) {
+      output = {"prediction": prediction};
+    } else {
+      if (self.tree.weighted) {
+        for (index = 0; index < len; index++) {
+          categoryMap[rootDistribution[index][0]] = 0.0;
         }
-        // distribution in the node
-        len = distribution.length;
+      } else {
+        // distribution in the root node
```

Comment:
I've checked the python bindings code and I  believe that in this case the initial value for instances is 1.0. Maybe this has to do with the differences Teresa found for non-weighted models recently.

Reply:
you're right, I forgot to alter the value here.